plugin/src/test/groovy/com/novoda/staticanalysis/internal/detekt/DetektIntegrationTest.groovy

```diff
@@ -190,4 +195,29 @@ class DetektIntegrationTest {
         assertThat(result.logs).doesNotContainLimitExceeded()
         assertThat(result.logs).doesNotContainDetektViolations()
     }
+
+    @Test
+    void shouldFailBuildWhenDetektConfiguredButNotApplied() {
+        def testProject = projectRule.newProject()
+                .withPenalty('''{
+                    maxWarnings = 0
+                    maxErrors = 0
+                }''')
+
+        def detektConfiguration = """
+        detekt { 
+            profile('main') { 
+                config = "${Fixtures.Detekt.RULES}" 
+                output = "${testProject.projectDir()}/build/reports"
+            }
+        }
+        """
+
+        testProject = testProject.withToolsConfig(detektConfiguration)
```

Comment:
more of a `nit:` but why extracting all these loval vars that bloat the test body? have you considered to extact a couple of methods in this class that would help to reuse the same setup across tests? (see `CheckstyleIntegrationTest` or similar when it comes to tool configuration)